app/models/services/UserService.scala

```diff
@@ -19,22 +19,21 @@ class UserService @Inject() (userDAO: UserDAO)(implicit ex: ExecutionContext) ex
   def save(user: User): Future[User] = userDAO.save(user)
 
   def save(profile: CommonSocialProfile): Future[User] = {
```

Comment:
Is there some pattern to destructure `profile` arg upfront so you don't have to couple it with `user`? The `match` below doesn't seem to actually depend on it--you're only matching on it to destructure it right? Maybe something like:
```scala
  def save(profile: CommonSocialProfile): Future[User] = profile match {
    case CommonSocialProfile(loginInfo, Some(first), Some(last), _, Some(email), _) =>
      userDAO.find(loginInfo).flatMap { user =>
        user match {
          // Assumes social profile (google) will have name and email
          case Some(u) =>
            userDAO.save(u.copy(firstName = first, lastName = last, email = email))
          case None =>
            userDAO.save(User(
              userID = 0,
              credentialId = loginInfo.providerID,
              firstName = first,
              lastName = last,
              userType = "user_type",
              email = email
            ))
        }
      }
  }
```